### PR TITLE
ENG-967 add get by email

### PR DESCRIPTION
**Pull request by i-mann**
This change is [Reviewable](https://reviewable.io/reviews/work180/auth0-management/5)

### CODE REVIEW

File: src/users/mod.rs [hunk 1]
```diff
@@ -114,6 +114,21 @@ impl UsersManager {
...
+  /// * `email` - The email of the user to retrieve.
+  /// # Scopes
+  /// * `read:users`
+  /// * `read:user_idp_tokens`
+  pub async fn get_by_email<A, U, S: AsRef<str>>(&self, email: S) -> Result<User<A, U>, Auth0Error>
+  where
+    A: DeserializeOwned + Send + Sync,
+    U: DeserializeOwned + Send + Sync,
+  {
+    GetUserByEmail::new(email).send_to(&self.0).await
```
Comment (tonisefton) [inline src/users/mod.rs:129]: >    mismatched types
  expected reference `&str`
found type parameter `S`rustc[E0308](https://doc.rust-lang.org/error-index.html#E0308)
mod.rs(117, 35): this type parameter
mod.rs(122, 5): arguments to this function are incorrect
user_get.rs(35, 10): associated function defined here

**Review by mterring — CHANGES_REQUESTED** [review]
> Reviewed 1 of 2 files at r2, 1 of 1 files at r3, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/work180/auth0-management/5)* status: all files reviewed, 2 unresolved discussions (waiting on @i-mann, @robmarto, and @tonisefton)

___
*[`src/users/mod.rs` line 129 at r1](https://reviewable.io/reviews/work180/auth0-management/5#-NHWWsInKY187gQ4RNs1-r1-129:-NHxcyeKAvlIcNDHtsiX:b-esknc9) ([raw file](https://github.com/work180/auth0-management/blob/c9d72721967f82288cdc1e8536fe06873f83d3d1/src/users/mod.rs#L129)):*
<details><summary><i>Previously, tonisefton (Toni Sefton) wrote…</i></summary><blockquote>

>    mismatched types
  expected reference `&str`
found type parameter `S`rustc[E0308](https://doc.rust-lang.org/error-index.html#E0308)
mod.rs(117, 35): this type parameter
mod.rs(122, 5): arguments to this function are incorrect
user_get.rs(35, 10): associated function defined here
</blockquote></details>

I suspect what you want to do is remove the generic and just type `email`:



_Suggestion:_
```XML
  pub async fn get_by_email<A, U>(&self, email: &str) -> Result<Vec<User<A, U>>, Auth0Error>
  where
    A: DeserializeOwned + Send + Sync,
    U: DeserializeOwned + Send + Sync,
  {
    GetUserByEmail::new(email).send_to(&self.0).await
```
___
*[`src/users/mod.rs` line 117 at r3](https://reviewable.io/reviews/work180/auth0-management/5#-NHxcNgC23dprh_yIhnb:-NHxcNgC23dprh_yIhnc:b-xez5v4) ([raw file](https://github.com/work180/auth0-management/blob/2ba6b5a0d4c60d484ae441056beda0ca234fd9a8/src/users/mod.rs#L117)):*
> ```XML
>   }
> 
>    /// Retrieve user details. A list of fields to include or exclude may also be specified.
> ```

Nit: extra space indent



_Code quote:_
```XML
  ///
```

**Review by i-mann — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/work180/auth0-management/5)* status: 1 of 2 files reviewed, 1 unresolved discussion (waiting on @mterring, @robmarto, and @tonisefton)

___
*[`src/users/mod.rs` line 129 at r1](https://reviewable.io/reviews/work180/auth0-management/5#-NHWWsInKY187gQ4RNs1-r1-129:-NHxmQT44ZYXaU7G1Z2p:b-896fix) ([raw file](https://github.com/work180/auth0-management/blob/c9d72721967f82288cdc1e8536fe06873f83d3d1/src/users/mod.rs#L129)):*
<details><summary><i>Previously, mterring (Michael Terrington) wrote…</i></summary><blockquote>

I suspect what you want to do is remove the generic and just type `email`:
</blockquote></details>

Done.

**Review by mterring — APPROVED** [review]
> Reviewed 1 of 1 files at r4, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/work180/auth0-management/5)* status: all files reviewed, 1 unresolved discussion (waiting on @robmarto and @tonisefton)

**Pull request merged**